AEPAssurance/Source/QuickConnect/QuickConnectManager.swift

```diff
@@ -0,0 +1,86 @@
+/*
+ Copyright 2022 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import Foundation
+import UIKit
+import AEPServices
+
+class QuickConnectManager {
+
+    private let stateManager: AssuranceStateManager
+    private let uiDelegate: AssurancePresentationDelegate
+    private let quickConnectService = QuickConnectService()
+    private let LOG_TAG = "QuickConnectManager"
+
+    init(stateManager: AssuranceStateManager, uiDelegate: AssurancePresentationDelegate) {
+        self.stateManager = stateManager
+        self.uiDelegate = uiDelegate
+    }
+
+    func createDevice() {
+        quickConnectService.shouldRetryGetDeviceStatus = true
+        guard let orgID = stateManager.getURLEncodedOrgID() else {
+            // log here
+            Log.debug(label: LOG_TAG, "orgID is unexpectedly nil")
+            return
+        }
+        quickConnectService.registerDevice(clientID: stateManager.clientID, orgID: orgID, completion: { error in
+            guard let error = error else {
+                self.checkDeviceStatus()
+                return
+            }
+            self.uiDelegate.quickConnectError(error: error)
+         })
+     }
+    
+    private func checkDeviceStatus() {
+        
+        guard let orgID = stateManager.getURLEncodedOrgID() else {
```

Comment:
Since `checkDeviceStatus()` has a possibility of being called more than once, at different times, it think it may be better for it use the `orgId` from the device creation call. In that way, in the rare possibility that the Configuration shared state has changed (remote config overwriting cached config) by the time of the next status check we will not end up checking status with the new orgId and result in a failure!

Thinking about it more, we might just want to see if we have to re start the quick connect flow in cases where Configuration state affects the device creation parameters. We can discuss and handle this at a later time though given the possibility of occurence.

Reply:
I think i like the idea of restarting the flow when the state changes. But i agree that we can discuss this more and do it at a later date.